src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -317,6 +321,48 @@ class Instance;
 IE_CORE_FORWARDDECLARE( ShaderCache );
 IE_CORE_FORWARDDECLARE( InstanceCache );
 IE_CORE_FORWARDDECLARE( LightListCache );
+IE_CORE_FORWARDDECLARE( ArnoldObject );
+
+class ArnoldLight;
+class ArnoldBlocker;
+
+/// The BlockerConnections can be used to record changes regarding connections
+/// between blockers and lights and to communicate them to Arnold.
+class BlockerConnections : public IECore::RefCounted
+{
+
+public :
+
+	BlockerConnections( bool ownConnections )
+		: m_ownConnections( ownConnections )
+	{
+	}
+
+	// Record changes. These can be called concurrently.
+	void registerLight( const std::string &lightName, ArnoldLight *light );
+	void deregisterLight( const std::string &lightName );
+	void registerBlocker( const std::vector<std::string> &lightNames, ArnoldBlocker *blocker );
+	void deregisterBlocker( ArnoldBlocker *blocker );
+
+	// Communicate with Arnold. This needs to get called non-concurrently before rendering.
+	void update();
+
+private:
+
+	struct Filters
+	{
+		ArnoldLight *light;
+		std::set<ArnoldBlocker*> blockers;
```

Comment:
`unordered_set`

Reply:
Sorry, I messed up this one.

The change to the member is here: d55c78e
Here's where the rest is changed: 60efade130e878cec576ef060570928cf96e01c2

Will squash everything in anyway once we're done.